src/hooks/useSwapCallback.tsx

```diff
@@ -294,15 +37,18 @@ export function useSwapCallback(
 ): { state: SwapCallbackState; callback: null | (() => Promise<string>); error: ReactNode | null } {
   const { account, chainId, library } = useActiveWeb3React()
```

Comment:
If possible, replace this implementation, as it duplicates lib/hooks/swap/useSwapCallback.

Instead, call into that callback:
```
  const deadline = useTransactionDeadline()
  const addTransaction = useTransactionAdder()
  /* some other app-specific state */
  const {state, callback: libCallback} = useLibSwapCallback(...)
  const callback = useCallback(() => libCallback().then((response) => addTransaction(...)), [...])
```

Reply:
ahh good idea